src/condor_tests/ornithology-docs/source/api/custom_plugins.rst

```diff
@@ -0,0 +1,213 @@
+additional details which are forward slash (``/``) separated e.g.
+``debug://command/(details)``. The details may be optional depending
+on the command. The following are the available main commands:
+
+*EXIT*
+    - **Format**: ``schema://exit/<value>``
+    - **Example**: ``debug://exit/1``
+    - **Description**: Informs the plugin to exit immediately with an
+      intended exit code specified in the extra details (immediately
+      following the command). The plugin will always exit when this
+      command is specified but may not exit with the specified number
+      provided the following reasons:
+
+      1. Any previously parse URL had an invalid encoding.
+      2. The provided value was negative (exit with the absolute value).
+      3. The provided value was out of the range of ``0`` - ``123``.
+         Will choose the closest viable exit code (likely ``123``).
+
+*SLEEP*
+    - **Format**: ``schema://sleep/<value>``
+    - **Example**: ``debug://sleep/20``
+    - **Description**: Informs the plugin to sleep for the specified time
+      immediately following the command.
+
+*SIGNAL*
+    - **Format**: ``schema://signal/<value>``
+    - **Example**: ``debug://signal/SIGKILL``
+    - **Description**: Inform the plugin to immediately send the signal
+      specified immediately after the command to itself. The available
+      signal values are denoted by `python signal module <https://docs.python.org/3/library/signal.html#signal.SIG_DFL>`_
+
+*SUCCESS* and *ERROR*
+    - **Format**: ``schema://command/(details)`` [Details are optional]
+    - **Examples**: ``debug://success``, ``debug://error/``, ``debug://error/general[foo=True;]``
+    - **Description**: Denotes whether a transferred file was successful or
+      failed due to an error. If any **error** commands are encoding in
+      the transfer URLs then the plugin will exit with ``1`` unless specified
+      otherwise or an internal failure occurred. Both of these commands
+      will write a file transfer result ClassAd to the output file. This
+      ClassAd will be constructed of default values and whatever further
+      ClassAd details are encoded in the additional details (see detail
+      encoding below).
+
+**Special Details Encoding**
+    Details about how to update the returned ClassAd for a transfer URL via
+    the plugin can be encoded in portions of the URL (``schema://command/portion/portion/portion``)
+    using specified keywords ('sub commands'). Any portion of the URL not
+    beginning with a designated keyword is ignored. Parsing of the encoding
+    happens left to right of the URL; meaning later keywords details take
+    precedent. The following keywords can be specified in URL portions:
+
+    *GENERAL*
+        - **Format**: ``general([details])`` (``[details]`` is optional).
+        - **Example**: ``debug://error/general``
+        - **Description**: Update/insert ClassAd attributes associated in
+          the main body of the result ad.
```

Comment:
Considering adding either "see below for details of details encoding" or an example with a single attribute encoded, or both.